packages/core/src/rum/instrumentation/resourceTracking/implementation/DatadogRumResource/ResourceReporter.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { DdRum } from '../../../../../foundation';
+import type { RUMResource } from '../../domain/interfaces/RumResource';
+
+import { createTimings } from './resourceTiming';
+
+type ResourceMapper = (resource: RUMResource) => RUMResource | null;
+
+export class ResourceReporter {
+    private mappers: ResourceMapper[];
+
+    constructor(resourceMappers: ResourceMapper[]) {
+        this.mappers = resourceMappers;
+    }
+
+    reportResource = (resource: RUMResource) => {
+        let modifiedResource: RUMResource | null = resource;
+
+        for (const mapper of this.mappers) {
+            modifiedResource = mapper(resource);
+            if (modifiedResource === null) {
+                return;
+            }
+        }
+
+        reportResource(modifiedResource);
```

Comment:
TypeScript noob question: how does method resolution works here? because there is `ResourceReporter#reportResource` and also `const reportResource` definition below. To call the former we need to use `this` receiver?

Reply:
yes, that's the way javascript works.